### PR TITLE
[1.1.1] Fix out-of-tree build failure

### LINKED ISSUE #11940: [1.1.1] Out-of-tree builds fail on `make update`
It seems like currently oot-builds succeed only if the build dir and the source dir are at the same level of nesting

### Steps to reproduce

```bash
; cd $osslrepo
; git checkout origin/OpenSSL_1_1_1-stable
; srcdir=$PWD
; blddir=../_build/nest/a/little/more
; mkdir -p $blddir
; cd $blddir
; $srcdir/config
; make update
```

Notice that if `$blddir` is set to `../_build`, the failure is avoided.

~~To quote @levitte: "I have a branch!" (where I can trigger the error in Travis): https://github.com/romen/openssl/tree/issues/1.1.1/oot_build~~

~~[Here](https://travis-ci.com/github/romen/openssl/builds/171098007) you can see the build passing using `../_build` as the build dir, [here](https://travis-ci.com/github/romen/openssl/builds/171097939) you can see the build failing using `../_build/nest/a/little/more` instead.~~
[Here](https://github.com/openssl/openssl/pull/14388/checks?check_run_id=2012180309) you can see the build failing in GitHub CI: #14388.

This seems to be related to #11184 and #11185, that addressed a similar defect in `master`.

_edit: updated to point to #14388 now that Travis CI is no more._

### PR DESCRIPTION
**Pull request by romen**
**Note: this PR is ready for review**

This was originally a PR to just trigger the issue described in #11940 , but I have reworked it to include the fix authored by @levitte and to keep the extra CI job (as it does not take too much time and this is not the first time we accidentally break out-of-tree builds).

## Checklist

- [x] Fixes #11940
- [x] Add CI job for #11940

<details>
<summary>Original description</summary>

# THIS PR SHOULD NOT BE MERGED

This PR is just meant to exercise the issue reported in #11940 using the new CI infrastructure, so that all it takes to verify if it is solved is to rebase this branch.

Note that one of the commits is dropping some jobs as they take longer and are not relevant for this issue.
</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,79 +48,35 @@ jobs:
...
+            echo "mysrcdir=$mysrcdir" | tee -a $GITHUB_ENV
+            myblddir=../_build/nest/a/little/more
+            mkdir -p $myblddir
+            echo "myblddir=$myblddir" | tee -a $GITHUB_ENV
       - name: config
-        run: CC=clang ./config --strict-warnings && perl configdata.pm --dump
+        run: |
+            set -eux
+            myblddir=${{ env.myblddir }}
+            mysrcdir=${{ env.mysrcdir }}
```
Comment (levitte) [inline .github/workflows/ci.yml:67]: Incidently, this was interesting to see.  I had no idea there was such a feature.
(and no, I don't need to have it explained, I got the connection with `$GITHUB_ENV`)

Comment (romen) [reply .github/workflows/ci.yml:67]: I discovered this today as well!

**Comment by levitte** [comment]
At this point, the "DO NOT MERGE" message isn't relevant any more, I really think this should become fully merged

**Comment by romen** [comment]
> At this point, the "DO NOT MERGE" message isn't relevant any more, I really think this should become fully merged

I was still working on editing the title and the description!

Now it is ready to review!

**Comment by romen** [comment]
Notice the `out-of-tree_build` joib only does the build, and intentionally omits `make test`, this is to avoid burdening the CI with longer running jobs that bring little benefit, as `make test` for that configuration is already run elsewhere.

**Comment by levitte** [comment]
> Notice the out-of-tree_build job only does the build, and intentionally omits make test, this is to avoid burdening the CI with longer running jobs that bring little benefit, as make test for that configuration is already run elsewhere.

`make test TESTS='0[0-9]'` should be small enough

**Comment by romen** [comment]
> make test TESTS='0[0-9]'

Done with bd6a6444b2

**Review by levitte — APPROVED** [review]

**Comment by romen** [comment]
Merged to `OpenSSL_1_1_1-stable` with:

- 732682aeb6 _[github-ci] Add a out-of-tree_build job_
- 01b9e18fb2 _[1.1.1] Fix `make update` for out-of-tree builds_

**Comment by romen** [comment]
Thanks @levitte for the feedback and the fix!

**Pull request closed without merging**